File: frontend/src/types/system.ts
export interface TenantRecord {
  id: number
  name: string
  code: string
  status: number
  remark: string
  createdAt: string
}

export interface UserRecord {
  id: number
  username: string
  nickname: string
  tenantId: number
  roleIds: number[]
  status: number
  createdAt: string
}

export interface RoleRecord {
  id: number
  name: string
  code: string
  tenantId: number
  menuIds: number[]
  status: number
  createdAt: string
}

export interface MenuRecord {
  id: number
  parentId: number
  title: string
  path: string
  component: string
  permission: string
  type: 'catalog' | 'menu' | 'button'
  sort: number
  visible: boolean
}

File: backend/internal/dto/system.go
package dto

type TenantDTO struct {
	ID        uint   `json:"id"`
	Name      string `json:"name"`
	Code      string `json:"code"`
	Status    int    `json:"status"`
	Remark    string `json:"remark"`
	CreatedAt string `json:"createdAt"`
}

type UserDTO struct {
	ID        uint   `json:"id"`
	TenantID  uint   `json:"tenantId"`
	Username  string `json:"username"`
	Password  string `json:"password,omitempty"`
	Nickname  string `json:"nickname"`
	RoleIDs   []uint `json:"roleIds"`
	Status    int    `json:"status"`
	CreatedAt string `json:"createdAt"`
}

type RoleDTO struct {
	ID        uint   `json:"id"`
	TenantID  uint   `json:"tenantId"`
	Name      string `json:"name"`
	Code      string `json:"code"`
	MenuIDs   []uint `json:"menuIds"`
	Status    int    `json:"status"`
	CreatedAt string `json:"createdAt"`
}

type MenuDTO struct {
	ID         uint   `json:"id"`
	ParentID   uint   `json:"parentId"`
	Title      string `json:"title"`
	Path       string `json:"path"`
	Component  string `json:"component"`
	Permission string `json:"permission"`
	Type       string `json:"type"`
	Sort       int    `json:"sort"`
	Visible    bool   `json:"visible"`
}
